snapcraft/cli/_errors.py

```diff
@@ -53,40 +67,52 @@ def exception_handler(exception_type, exception, exception_traceback, *,
 
     exit_code = 1
     is_snapcraft_error = issubclass(exception_type, errors.SnapcraftError)
-
-    if debug:
+    is_raven_setup = RavenClient is not None
+
+    # The use cases:
+    # - a non snapcraft handled error occurs and raven is setup,
+    #   so we go over confirmation logic showing the relevant traceback
+    # - a non snapcraft handled error occurs and raven is not setup,
+    #   so we just show the traceback
+    # - a snapcraft handled error occurs, --debug is set so a traceback
+    #   is shown
+    # - a snapcraft handled error occurs, --debug not set so only the
+    #   exception message is shown
```

Comment:
Docs similar to this are already in the docstring of this function. Perhaps this should take the form of an update to that docstring?